src/main/java/bio/overture/ego/service/VisaService.java

```diff
@@ -81,9 +78,22 @@ public List<Visa> getByTypeAndValue(@NonNull String type, @NotNull String value)
     }
   }
 
-  public void delete(@NonNull UUID id) {
-    checkExistence(id);
-    super.delete(id);
+  public List<Visa> getByTypeAndValueForPassport(@NonNull String type, @NotNull String value) {
+    val result = visaRepository.getByTypeAndValue(type, value);

```

Comment:
this method logic looks duplicated.. is it possible to reuse it and call it from getByTypeAndValue method?